### PR TITLE
Add hooks

### LINKED ISSUE #491: Convention for import statement ordering
Jonathan and I have been working together to try to define a ruleset for how import statements should be ordered, as he is developing a script to automate the sorting of these (#488 #487 #484).

The import order that we are currently proposing is the following.
1. public import statements
2. non-public Agda primitive imports
3. non-public import statements from the same namespace
4. non-public import statements from other namespaces
5. other `open` statements (this seems to be unconventional, but alas there are instances of this in the library)

Within these categories, the statements should be sorted alphabetically.

The reasoning is that this will hopefully correlate well with the relevance of each individual import, while still being somewhat intuitive to the user/contributor.

### PR DESCRIPTION
**Pull request by jonaprieto**
This PR adds pre-commit hooks to fix/enforce some conventions like automatically formatting the Agda import block insider `<details>` block.

- Closes #491
- Add infrastructure for #448

The hooks are written in Python for ease (also for the CI in the future) and are run using [pre-commit](https://pre-commit.com/).

The idea is once you have
staged all your changes, you can run (which by default also runs `make check`):

```bash
make pre-commit
```

It will automatically fix any formatting issues.

In principle, we could also add checks for other conventions in the future, like the ones
described in CONVENTIONS.md

Oh. I also changed CITATION.md as Github encourages to use the .cff extension. 

- [x] Update the CI to run pre-commit

### CODE REVIEW

File: CONVENTIONS.md [hunk 1]
```diff
@@ -29,3 +29,5 @@ Ideally, the first section of a file explains the idea, then proceeds to give th
       This will be displayed as [UniMath/agda-unimath](https://github.com/UniMath/agda-unimath).
 
 A good reference file for the expected structure of a file is [`foundation.cantor-schroder-bernstein-escardo`](https://raw.githubusercontent.com/UniMath/agda-unimath/master/src/foundation/cantor-schroder-bernstein-escardo.lagda.md).
+
+Please note that some conventions above are checks enforced by our pre-commit hooks, read more about them in [HOWTO-INSTALL.md](HOWTO-INSTALL.md#pre-commit-hooks).
```
Comment (fredrik-bakke) [inline CONVENTIONS.md:33]: "Please note that some of these conventions, but not all, are enforced by our pre-commit hooks[...]"

File: HOWTO-INSTALL.md [hunk 2]
```diff
@@ -100,3 +100,21 @@ Menlo, Source Code Pro, Consolas, Monaco, Lucida Console, Liberation Mono, DejaV
...
+```shell
+python3 -pip install pre-commit
+```
+
+Once you have installed `pre-commit`, next time before you open a new PR, please stage
+all your changes and run the following command:
+
+```shell
+make pre-commit
+```
```
Comment (fredrik-bakke) [inline HOWTO-INSTALL.md:120]: Sounds to me like this should be added to the `.nix` file. @VojtechStep

File: hooks/README.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+```
+pip install pre-commit
+```
+
+Next time you commit, pre-commit will run the hooks.
+
+```
+make pre-commit
+```
```
Comment (fredrik-bakke) [inline hooks/README.md:14]: Nix file?

Comment (jonaprieto) [reply hooks/README.md:14]: Oh, that instruction is actually if you ran `pre-commit install` before. Srry

**Pull request merged**